javalanglib/src/main/scala/java/lang/Math.scala

```diff
@@ -46,8 +49,22 @@ object Math {
 
   @inline def exp(a: scala.Double): scala.Double = js.Math.exp(a)
   @inline def log(a: scala.Double): scala.Double = js.Math.log(a)
-  @inline def log10(a: scala.Double): scala.Double = log(a) / 2.302585092994046
-  @inline def log1p(a: scala.Double): scala.Double = log(a + 1)
+
+  @inline def log10(a: scala.Double): scala.Double = {
+    if (assumingES6 || !js.isUndefined(g.Math.log10))
+      js.Math.log10(a)
+    else
+      log(a) / 2.302585092994046
+  }
+
+  @inline def log1p(a: scala.Double): scala.Double = {
+    if (assumingES6 || !js.isUndefined(g.Math.log1p))
+      js.Math.log1p(a)
+    else a match {
+      case -0.0 => -0.0
```

Comment:
This will also match `+0` because they are `==`. If the expected result for `+0` is `+0`, then a better way to write it is `if (a == 0.0) a else ...` which preserve the sign of `a`. (I'd recommend against using the `match` notation precisely for that reason: it's very confusing with zeros and NaNs.)